fog/view/server/src/router_request_handler.rs

```diff
@@ -49,11 +50,24 @@ where
 
         match result {
             Ok(response) => responses.send((response, WriteFlags::default())).await?,
-            Err(rpc_status) => return responses.fail(rpc_status).await,
+            Err(rpc_status) => {
+                log::error!(logger, "error handling request: {}", &method_name);
+                CONNECTED_CLIENTS.dec();
+                return responses.fail(rpc_status).await;
+            }
+        }
+    }
+    match responses.close().await {
+        Ok(value) => {
+            CONNECTED_CLIENTS.dec();
```

Comment:
Looks like the `dec()` happens in both cases so it should just happen before the match statement